src/IQFeed.CSharpApiClient/Lookup/Common/BaseLookupMessageHandler.cs

```diff
@@ -36,5 +57,29 @@ protected static MessageContainer<T> ProcessMessages<T>(Func<string, T> parser,
 
             return new MessageContainer<T>(parsedMessages, endMsg);
         }
+        
+        private static bool IsErrorMessage(string message)
+        {
+            // detecting messages like - "E,ERROR_TEXT" or "E,ERROR_TEXT,"
+            // NOTE: the regex pattern matching is needed to avoid problems like detecting a valid message like
+            // E,7,1,ENI, from search symbol request as an error. 
```

Comment:
the error pattern is in fact E,!NO_DATA!,, so this might simplify this thing a lot.

Reply:
E,!

Reply:
just realized this is not correct because message like this.
E,Invalid number of parameters.,

the pattern is then:
E,"MESSAGE"
!ENDMSG!,

Reply:
so I have the pattern...
we need to do a messages.length == 2
then we check message[0].split().length = 2
finally check message[1] == !ENDMSG!

if have an error.

Reply:
no need any regex and complicated code.

Reply:
@mrut2pac process with the rest of my comments. Ill be adjusting the ProcessMessages and providing unit tests.

Reply:
It's not that simple. Take a look at the documentation:
http://www.iqfeed.net/dev/api/docs/Level1viaTCPIP.cfm
here we have E,[ERROR TEXT]<CR><LF>
http://www.iqfeed.net/dev/api/docs/HistoricalviaTCPIP.cfm 
E,!SYNTAX_ERROR!,<CR><LF>
Or, if the servers return an error, you will receive an error in the following format:
E,[Error Text],<CR><LF>!ENDMSG!,<CR><LF>
If you specified a REQUESTID, it will be prepended to one of the above messages. For example:
REQUESTID,E,[Error Text],<CR><LF>
plus another undocumented case when we get 
E,!NO_DATA!,, with two ending commas
I have also seen E,'incorrect number of parameters' or something like that

Keep in mind the regex kicks in very rarely when we actually have an error and since it is not the main flow, I wouldn't worry about slightly heavier check to gain code simplicity...